rasa/utils/tensorflow/models.py

```diff
@@ -575,9 +575,13 @@ def _prepare_ffnn_layer(
         )
 
     def _prepare_dot_product_loss(
-        self, name: Text, scale_loss: bool, prefix: Text = "loss"
+        self,
+        name: Text,
+        scale_loss: bool,
+        prefix: Text = "loss",
+        loss_layer: tf.keras.layers.Layer = layers.DotProductLoss,
     ) -> None:
-        self._tf_layers[f"{prefix}.{name}"] = layers.DotProductLoss(
+        self._tf_layers[f"{prefix}.{name}"] = loss_layer(
```

Comment:
Not sure if that's the correct way because that would make it mandatory for any class inheriting from `TransformerModel` to have a function named `_prepare_dot_product_loss`. Right now if we keep it this way, only those models which do not use `SingleLabelDotProductLoss` but a variant of it need to call the base class's function with the correct argument.